File: Shopping-Service/src/main/java/in/neelesh/online/shopping/service/WishlistService.java
package in.neelesh.online.shopping.service;

import java.util.List;

import in.neelesh.online.shopping.dto.WishlistResponseDto;

public interface WishlistService {

	void addProductToWishlist(String realm, String customerId, String productId);

	void removeProductFromWishlist(String realm, String customerId, String productId);

	WishlistResponseDto getWishlistByCustomerId(String realm, String customerId);
	
	public void removeOrderedProductFromWishlist(String customerId, List<String> orderedProductId);

}

File: src/app/services/wishlist-service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { TokenService } from './token-service';

const base_url = 'http://localhost:1236/Shopping/wishlist';

export interface ProductResponseDto {
  id: string;
  name: string;
  price: number;
  image: string; 
}

export interface WishlistResponseDto {
  wishlistId: string;
  customerId: string;
  products: ProductResponseDto[];
}

@Injectable({
  providedIn: 'root'
})
export class WishlistService {
  base_url: any;
  baseUrl: any;
  realm: any;

  constructor(private http: HttpClient,
    private tokenService: TokenService
  ) { }

    private getAuthHeaders(): HttpHeaders {
    const token = this.tokenService.getToken() || '';
    return new HttpHeaders({
      Authorization: `Bearer ${token}`
    });
  }

  getWishlist(): Observable<WishlistResponseDto> {
    return this.http.get<WishlistResponseDto>(`${base_url}`, {
          headers: this.getAuthHeaders()
        });
  }

  removeFromWishlist(productId: string): Observable<void> {
  return this.http.delete<void>(`${base_url}/${productId}`, {
    headers: this.getAuthHeaders()
  });
  }
}
